keyboards/darkproject/kd83a_bfg_edition/kd83a_bfg_edition.c

```diff
@@ -0,0 +1,172 @@
+    {0, CS13_SW4,  CS14_SW4,  CS15_SW4},  //  63, k34, C
+    {0, CS13_SW5,  CS14_SW5,  CS15_SW5},  //  64, k44, V
+    {0, CS13_SW6,  CS14_SW6,  CS15_SW6},  //  65, k45, B
+    {0, CS13_SW7,  CS14_SW7,  CS15_SW7},  //  66, k55, N
+    {0, CS13_SW8,  CS14_SW8,  CS15_SW8},  //  67, k54, M
+    {0, CS13_SW9,  CS14_SW9,  CS15_SW9},  //  68, k64, ,
+    {0, CS13_SW10, CS14_SW10, CS15_SW10}, //  69, k74, .
+    {0, CS13_SW11, CS14_SW11, CS15_SW11}, //  70, k85, /
+    {1, CS4_SW8,   CS5_SW8,   CS6_SW8},   //  71, k91, Shift_R
+    {1, CS4_SW9,   CS5_SW9,   CS6_SW9},   //  72, k35, Up
+    {1, CS4_SW7,   CS5_SW7,   CS6_SW7},   //  73, k25, PgDn
+
+    {0, CS16_SW1,  CS17_SW1,  CS18_SW1},  //  74, k06, Ctrl_L
+    {0, CS16_SW2,  CS17_SW2,  CS18_SW2},  //  75, k90, Win_L
+    {0, CS16_SW3,  CS17_SW3,  CS18_SW3},  //  76, k93, Alt_L
+    {0, CS16_SW6,  CS17_SW6,  CS18_SW6},  //  77, k94, Space
+    {0, CS16_SW9,  CS17_SW9,  CS18_SW9},  //  78, k95, Alt_R
+    {0, CS16_SW10, CS17_SW10, CS18_SW10}, //  79, k92, FN
+    {0, CS16_SW12, CS17_SW12, CS18_SW12}, //  80, k04, Ctrl_R
+    {1, CS4_SW10,  CS5_SW10,  CS6_SW10},  //  81, k03, Left
+    {1, CS4_SW11,  CS5_SW11,  CS6_SW11},  //  82, k73, Down
+    {1, CS10_SW5,  CS11_SW5,  CS12_SW5},  //  83, k05, Right
+
+    {1, CS10_SW10, CS11_SW10, CS12_SW10}, //  84, kb0, Z1
+    {1, CS10_SW11, CS11_SW11, CS12_SW11}, //  85, kb1, Z2
+};
+
+#endif
+
+#ifdef EEPROM_ENABLE
+
+#include "spi_master.h"
+
+void spi_init(void) {
+    static bool is_initialised = false;
+    if (!is_initialised) {
+        is_initialised = true;
+
+        // Try releasing special pins for a short time
+        setPinInput(SPI_SCK_PIN);
+        setPinInput(SPI_MOSI_PIN);
+        setPinInput(SPI_MISO_PIN);
+
+        chThdSleepMilliseconds(10);
+
+        palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), PAL_MODE_ALTERNATE(SPI_SCK_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST | PAL_WB32_CURRENT_LEVEL3);
+        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), PAL_MODE_ALTERNATE(SPI_MOSI_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), PAL_MODE_ALTERNATE(SPI_MISO_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+    }
+}
+
+#endif
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(C0);
+    setPinOutput(C15);
+    keyboard_pre_init_user();
+};
+void housekeeping_task_kb(void) {
+    writePin(C15, keymap_config.no_gui);
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+
+    switch (keycode) {
+
+        case KC_SIRI:
+            if (record->event.pressed) {
+                host_consumer_send(0xCF);
+            } else {
+                host_consumer_send(0);
+            }
+            return false; /* Skip all further processing of this key */
+        default:
+            return true; /* Process all other keycodes normally */
+    }
+};
```

Comment:
Defines a default behaviour for encoders that works for keymaps compiled in QMK Configurator. (Configurator doesn't use `ENCODER_MAP`.)

```suggestion
};

#if defined(ENCODER_ENABLE)
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
    if (clockwise) {
        tap_code(KC_VOLU);
    } else {
        tap_code(KC_VOLD);
    }
    return false;
};
#endif
```